### PR TITLE
Refactor options and worker dependencies

### PR DESCRIPTION
**Pull request by meiliang86**
This is the first part of a general options and worker dependencies cleanup. After this refactoring, user should follow the sequence below to setup worker and its dependencies:
- IWorkflowService is a client stub for communicating to Cadence Server
- WorkflowClient wraps IWorkflowService and provides higher level API to start, signal, query and wait for completion of workflows.
- WorkerFactory wraps WorkflowClient and can be used to create instances of Workers.
- Worker listens on a single task list and is created by a WorkerFactory.

Ref: https://github.com/temporalio/sdk-java/pull/32 and https://github.com/temporalio/sdk-java/pull/33

### CODE REVIEW

File: src/main/java/com/uber/cadence/internal/sync/TestWorkflowEnvironmentInternal.java [hunk 1]
```diff
@@ -17,7 +17,76 @@
 
 package com.uber.cadence.internal.sync;
 
-import com.uber.cadence.*;
```
Comment (vytautas-karpavicius) [inline src/main/java/com/uber/cadence/internal/sync/TestWorkflowEnvironmentInternal.java:20]: Any reason why this as expanded? Or is this auto-formated?

Comment (meiliang86) [reply src/main/java/com/uber/cadence/internal/sync/TestWorkflowEnvironmentInternal.java:20]: The convention is to always expand the imports, so ppl can check what exactly is imported. That said the IDE sometimes automatically collapses the import and that's why you see starts in some place.

File: src/main/java/com/uber/cadence/worker/Worker.java [hunk 2]
```diff
@@ -660,24 +600,6 @@ public synchronized void shutdown() {
...
         worker.shutdown();
       }
-      closeServiceWhenTerminated();
-    }
-
-    /**
-     * Closes Cadence client object. It should be closed only after all tasks have completed
-     * execution as tasks use it to report completion.
-     */
-    private void closeServiceWhenTerminated() {
```
[the remark this answers is not in the document]
Comment (meiliang86) [reply src/main/java/com/uber/cadence/worker/Worker.java:670]: This is intended. workflow service is initialized before worker, and should be closed after.
Or the intention is to decouple workflow service lifecycle management from worker lifecycle management.

[the remark this answers is not in the document]
Comment (meiliang86) [reply src/main/java/com/uber/cadence/worker/Worker.java:670]: This is initialized in the application startup logic, and should be teared down in the application shutdown logic.

**Review by mkolodezny — APPROVED** [review]

File: src/main/java/com/uber/cadence/worker/Worker.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (vytautas-karpavicius) [reply src/main/java/com/uber/cadence/worker/Worker.java:670]: Ok, thank you for explanation.

**Review by vytautas-karpavicius — APPROVED** [review]

**Pull request merged**